File: src/main/java/com/example/play/AloneSugoroku.java
package com.example.play;

import java.util.*;

public class AloneSugoroku {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String[] line = sc.nextLine().split(" ");
        Map<Integer,String> dice = new HashMap<>();
        dice.put(Integer.parseInt(line[0]), "T");
        dice.put(Integer.parseInt(line[1]), "B");
        dice.put(Integer.parseInt(line[2]), "U");
        dice.put(Integer.parseInt(line[3]), "D");
        dice.put(Integer.parseInt(line[4]), "L");
        dice.put(Integer.parseInt(line[5]), "R");

        int n = Integer.parseInt(sc.nextLine());
        ArrayList<Integer> boardList = new ArrayList<>();

        for (int i = 0; i < n; i++){
            boardList.add(Integer.parseInt(sc.nextLine()));
        }

        String currentStr = "T";
        int count = 0;

        for (int i = 1; i < boardList.size(); i++){
            int nextNum = boardList.get(i);
            String nextStr = dice.get(nextNum);
            if ((currentStr.equals("T") && nextStr.equals("B"))
                || (currentStr.equals("B") && nextStr.equals("T"))
                || (currentStr.equals("L") && nextStr.equals("R"))
                || (currentStr.equals("R") && nextStr.equals("L"))
                || (currentStr.equals("U") && nextStr.equals("D"))
                || (currentStr.equals("D") && nextStr.equals("U"))) {
                count += 2;
                currentStr = nextStr;
            } else if (!currentStr.equals(nextStr)) {
                count++;
                currentStr = nextStr;
            }
        }
        System.out.println(count);
    }
}

File: src/main/java/com/example/play/AloneSugoroku.kt
package com.example.play

fun main(args: Array<String>) {
    val line = readLine()?.split(" ") ?: return
    val dice = mapOf(
            line[0].toInt() to "T",
            line[1].toInt() to "B",
            line[2].toInt() to "U",
            line[3].toInt() to "D",
            line[4].toInt() to "L",
            line[5].toInt() to "R"
    )

    val masu = readLine()?.toInt() ?: return
    val boardList = mutableListOf<Int>()

    for (num in 0 until masu) {
        boardList.add(readLine()?.toInt() ?: return)
    }

    var currentStr = "T"
    var count = 0

    for (num in 1 until boardList.size) {
        val nextStr = dice[boardList[num]] ?: return
        if ((currentStr == "T" && nextStr == "B")
                || (currentStr == "B" && nextStr == "T")
                || (currentStr == "L" && nextStr == "R")
                || (currentStr == "R" && nextStr == "L")
                || (currentStr == "U" && nextStr == "D")
                || (currentStr == "D" && nextStr == "U")) {
            count += 2
            currentStr = nextStr
        } else if (currentStr != nextStr) {
            count += 1
            currentStr = nextStr
        }
}
    println(count)
}